addons/l10n_pe_website_sale/static/src/js/website_sale.js

```diff
@@ -0,0 +1,95 @@
+/** @odoo-module **/
+import {WebsiteSale} from "@website_sale/js/website_sale";
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        "change select[name='state_id']": "_onChangeState",
+        "change select[name='city_id']": "_onChangeCity",
+    }),
+    start: function () {
+        this.selectCities = document.querySelector("select[name='city_id']");
+        this.selectDistricts = document.querySelector("select[name='l10n_pe_district']");
+        this.cityBlock = document.querySelector(".div_city");
+        this.autoFormat = document.querySelector(".checkout_autoformat");
+        this.selectState = document.querySelector("select[name='state_id']");
+        this.isPeruvianCompany = parseInt(document.querySelector("html").getAttribute("data-is-peruvian-company"));
+        if(this.isPeruvianCompany && this.selectState) {
+            let selectCountry = document.querySelector("select[name='country_id']").value;
+            if (this.selectState.value == "" && selectCountry !== '') {
+                this.selectState.options[1].selected = true;
+            }
+        }
```

Comment:
Handle the delivery address issue by fixing `_onChangeState` (see below)
```suggestion
        this.selectCountry = document.querySelector("select[name='country_id']");
        this.isPeruvianCompany = this.selectCountry?.dataset.company_country_code === 'PE';
```